src/components/structures/TimelinePanel.js

```diff
@@ -245,6 +245,69 @@ var TimelinePanel = React.createClass({
         }
     },
 
+    _unpaginateEvents: function(backwards) {
+        // Do not unpaginate below timelineCap
+        if (this.state.events.length < this.props.timelineCap) {
+            return;
+        }
+
+        let dir = backwards ? EventTimeline.BACKWARDS : EventTimeline.FORWARDS;
+        debuglog("TimelinePanel: unpaginating events in direction", dir);
+        // The number of events to unpaginate
+        let count = 0;
+        // The parent (ScrollPanel) of event tiles including MemberEventListSummary elements
+        let eventTileContainer = this.refs.messagePanel.getEventTileContainer();
+        let tiles = ReactDOM.findDOMNode(eventTileContainer).children || [];
+
+        // No tiles rendered; no unpagination required
+        if (tiles.length === 0) {
+            return;
+        }
+
+        // Count the events for each tile, with a default of one per element, but for any
```

Comment:
so this looks like it probably works, but the idea of the scrollpanel design is that it tries to abstract away concerns about element heights.

Instead I would suggest hooking into the system used to relate the scroll position back to events (the scroll token). ScrollPanel could calculate the scroll token of the first node in the section to be unpaginated; TimelinePanel could turn that back into an event id and do the unpagination based on that. I think that could also save you having to mess about with `data-number-events`.